Mus 94 refactor piano view

AndrejMueller01 opened the pull request.

AndrejMueller01 committed:
MUS-94: Added a Piano View based on xml Files.

Started to make a dynamic Piano View. Button Positioning

AndrejMueller01 committed:
MUS-94: Added NoteSheetView, Refactored PianoView

Creating the basic layout with Fragments and XML files.

AndrejMueller01 committed:
Merge commit

AndrejMueller01 committed:
MUS-94: NoteSheetView, Refactoring

fixing NoteSheetView, renamed Key and Insrument classes, fixed
TickThread

AndrejMueller01 committed:
MUS-94: Rename classes

Instument -> MusicalInstrument
Key -> MusicalKey

AndrejMueller01 committed:
MUS-86: Export as midi

created menu, added first save function (saving midi to:
sdcard/musicdroid/test.midi)

Commit:
Revert "MUS-86: Export as midi"

This reverts commit cd50cc471b5ed56f357d3a5361930fa141424f6f.

Commit:
MUS-86: Refactoring

Added AbstractTickThread, Moved Project class, removed name attribute of
Project, refactored Converter classes around this change

Commit:
MUS-94: Note Tests

Started adding old note tests. They run, but there is still some
refactoring to do.

Commit:
MUS-94: Note Tests Round 2

Refactoring

Commit:
MUS-94: Note Tests Round 3

Commit:
MUS-94: Note Tests Round 4

Refactoring + MIDI Tests (work in progress)

Commit:
MUS-94: Note Tests Round 5

added symbol tests

Commit:
MUS-94: Midi packaged moved

Moved midi package into note package. Deleted the Midi Test Case (with
todo mark to add it again soon)

Commit:
MUS-94: Midi package moved

deleted files because of move

AndrejMueller01 committed:
MUS-94: TestCases for PianoViewFragment

Commit:
MUS-94: TestCases for Piano

Refactoring / Code Review with Andrej of Test Cases (PianoActivityTest
and PianoKeyTest)

Commit:
MUS-94: TestCases for Piano

Changed a single parameter in the PianoKeyTest class. See Diff, its
super easy

Commit:
MUS-94: InstrumentActivity

Fixed default values in InstrumentActivity. Now the MusicalKey and the
MusicalInstrument gets set by the child classes (like the PianoActivity)

Commit:
MUS-94: deleting classes

deleted all classes which were not used in this branch so that we can
keep the code nice

AndrejMueller01 committed:
MUS-94: testcase fix

Added an annotation to testCalculatePianoKeyPositions()

AndrejMueller01 committed:
MUS-94: ui testcase fix

refactoring testPianoKeys()

Commit:
MUS-94: refactoring after code review

Changed names of a few variables and methods in the PianoViewFragment
after a code review by Bianca

AndrejMueller01 committed:
MUS-94: small refactoring

SimonStefan commented:
Your pull request isn't mergeable. Please fix it.
Further improvements:
- don't change the following files. Which program are you using for git? Have you set up your working environment correctly? [1]
  - .idea/encodings.xml
  - .idea/misc.xml
  - Musicdroid.iml
  - musicdroid/musicdroid.iml
  - musicdroidSourceTest/musicdroidSourceTest.iml
- don't put jar-Files into the repository if they are available on Maven Central!! [2]
- this pull request is far too big. And why are there e.g. break, cross and tact images in there? I think at the planning game these features where not planned.

[1] https://github.com/Catrobat/Musicdroid/wiki
[2] http://search.maven.org/#artifactdetails%7Ccom.jayway.android.robotium%7Crobotium%7C5.2.1%7Cpom

bteufl commented:
Why is the very first pull request not mergeable?

bteufl commented on musicdroid/src/main/java/org/catrobat/musicdroid/pocketmusic/instrument/piano/PianoViewFragment.java:
```diff
@@ -0,0 +1,208 @@
+
+        setBlackKeyInvisible(Octave.DEFAULT_INACTIVE_BLACK_KEY);
+
+        // TODO: optimize touch events
```
Why is there a todo?

bteufl commented:
The break-images and the tact-image is not needed... In fact, no picture should be needed within the task refactor piano-view...

AndrejMueller01 committed:
MUS-94: Removing pngs, restoring xml and iml files

Removed some unused pictures, trying to fix encodings, misc and iml
files

AndrejMueller01 committed:
MUS-94: gitignore update

for some reason the .gitignore file is not working correctly

SimonStefan commented:
You don't have to close a pull request if there are suggested improvements. You could just push the improvements to the same branch and the pull request will be updated automatically.

To clarify my comment: don't ever change the files I've mentioned

AndrejMueller01 committed:
MUS-94: preparing for pullrequest 1

used maven for jenkins lib, restoring files (iml, xml)

AndrejMueller01 committed:
Update Musicdroid.iml

AndrejMueller01 commented:
Ok.

AndrejMueller01 committed:
Update encodings.xml

adding newline at the end of file

AndrejMueller01 committed:
Update misc.xml

adding newline at the end

AndrejMueller01 committed:
Update Musicdroid.iml

adding newline at the end

AndrejMueller01 committed:
Update musicdroidSourceTest.iml

adding newline at the end

AndrejMueller01 committed:
Refactoring and bug fix

Jenkins Lint Issues: removed some hardcoded stuff, new structure of
layout file, wrong value: DEFAULT_INACTIVE_BLACK_KEY

AndrejMueller01 committed:
MUS-94: testcase fix

fixed some tcs: button visibility

fwinkelbauer commented:
https://jenkins.catrob.at/view/All-Categories/view/Musicdroid/job/Musicdroid-Custom-Branch-Gradle-Emulator/44/

SimonStefan commented on .gitignore:
```diff
@@ -23,14 +23,11 @@ local.properties
 proguard/
 
 # IntelliJ config files
-/.idea/workspace.xml
```
Please do not change the .gitignore! I played around at least 10 hours to find out the best solution for about 100 students using shared settings and different platforms. There is an instruction of the project setup in the Github Wiki you should have read in the first place. Now please do the following:
- undo the changes to .gitignore and commit it
- delete all unused branches! (there could be huge problems!)
- close Android Studio!
- execute in Terminal in project root (all written code should be committed, or will be lost)
  - "git reset --hard" should reset the iml-files and misc.xml
  - "git clean -xdf" should delete some hidden settings files
- proceed last 3 steps: https://github.com/Catrobat/Musicdroid/wiki

PS: in the current version of Android Studio Google fixed a bug. So maybe there will be an updated version of these settings without the iml-files soon.

AndrejMueller01 committed:
Undo changes: .gitignore
